File: backend/internal/model/standings.go
package model

import "github.com/google/uuid"

// StandingsEntry represents a single entry in the league standings
type StandingsEntry struct {
	Rank            int       `json:"rank"`
	ParticipantID   uuid.UUID `json:"participant_id"`
	UserID          uuid.UUID `json:"user_id"`
	DriverName      string    `json:"driver_name"`
	TeamName        *string   `json:"team_name,omitempty"`
	TotalPoints     float64   `json:"total_points"`
	RacePoints      float64   `json:"race_points"`
	SprintPoints    float64   `json:"sprint_points"`
	Wins            int       `json:"wins"`
	Podiums         int       `json:"podiums"`
	FastestLaps     int       `json:"fastest_laps"`
	DNFs            int       `json:"dnfs"`
	RacesCompleted  int       `json:"races_completed"`
}

// TeamStandingsEntry represents a single team entry in the standings
type TeamStandingsEntry struct {
	Rank           int     `json:"rank"`
	TeamName       string  `json:"team_name"`
	TotalPoints    float64 `json:"total_points"`
	RacePoints     float64 `json:"race_points"`
	SprintPoints   float64 `json:"sprint_points"`
	Wins           int     `json:"wins"`
	Podiums        int     `json:"podiums"`
	FastestLaps    int     `json:"fastest_laps"`
	DNFs           int     `json:"dnfs"`
	DriverCount    int     `json:"driver_count"`
}

// LeagueStandingsResponse represents the response for league standings
type LeagueStandingsResponse struct {
	LeagueID      uuid.UUID            `json:"league_id"`
	LeagueName    string               `json:"league_name"`
	Season        int                  `json:"season"`
	TotalRaces    int                  `json:"total_races"`
	Standings     []StandingsEntry     `json:"standings"`
	TeamStandings []TeamStandingsEntry `json:"team_standings"`
}

File: frontend/src/services/standings.ts
import api from './api'

export interface StandingsEntry {
  rank: number
  participant_id: string
  user_id: string
  driver_name: string
  team_name?: string
  total_points: number
  race_points: number
  sprint_points: number
  wins: number
  podiums: number
  fastest_laps: number
  dnfs: number
  races_completed: number
}

export interface TeamStandingsEntry {
  rank: number
  team_name: string
  total_points: number
  race_points: number
  sprint_points: number
  wins: number
  podiums: number
  fastest_laps: number
  dnfs: number
  driver_count: number
}

export interface LeagueStandingsResponse {
  league_id: string
  league_name: string
  season: number
  total_races: number
  standings: StandingsEntry[]
  team_standings: TeamStandingsEntry[]
}

export const standingsService = {
  async getByLeague(leagueId: string): Promise<LeagueStandingsResponse> {
    const response = await api.get<LeagueStandingsResponse>(`/leagues/${leagueId}/standings`)
    return response.data
  },
}

export default standingsService
